### PR TITLE
Add new Alluxio-FUSE as local cache solution with UFS gateway

### PR DESCRIPTION
**Pull request by LuQQiu**
Add new Alluxio-FUSE as local cache solution with UFS gateway.
Translates under storage sematic to POSIX API semantic.
Provides optional local data/metadata cache.
New Alluxio-FUSE allows users to launch a process directly to connect to their desired under storage instead of Alluxio system cache (master & workers).

### CODE REVIEW

File: core/client/fs/src/main/java/alluxio/client/file/FileSystem.java [hunk 1]
```diff
@@ -161,7 +162,9 @@ public static FileSystem create(FileSystemContext context) {
     public static FileSystem create(FileSystemContext context, FileSystemOptions options) {
       AlluxioConfiguration conf = context.getClusterConf();
       checkSortConf(conf);
-      FileSystem fs = new BaseFileSystem(context);
+      FileSystem fs = options.getUfsFileSystemOptions().isPresent()
```
Comment (yyongycy) [inline core/client/fs/src/main/java/alluxio/client/file/FileSystem.java:165]: can this be used for a normal alluxio client?

File: integration/fuse/src/main/java/alluxio/fuse/options/FuseOptions.java [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+   * @param fileSystemOptions the file system options
+   * @param updateCheckEnabled whether to enable update check
+   * @return the file system options
+   */
+  public static FuseOptions create(AlluxioConfiguration conf,
+      FileSystemOptions fileSystemOptions, boolean updateCheckEnabled) {
+    Set<String> mountOptions = conf.getList(PropertyKey.FUSE_MOUNT_OPTIONS)
+        .stream().filter(a -> !a.isEmpty()).collect(Collectors.toSet());
+    LibfuseVersion version = AlluxioFuseUtils.getLibfuseVersion(conf);
+    if (!conf.getBoolean(PropertyKey.FUSE_JNIFUSE_ENABLED)
```
Comment (yyongycy) [inline integration/fuse/src/main/java/alluxio/fuse/options/FuseOptions.java:71]: still support jnr fuse?

**Review by yyongycy — APPROVED** [review]
LGTM

File: integration/fuse/src/main/java/alluxio/fuse/options/FuseOptions.java (hunk 2 above)
Comment (LuQQiu) [reply integration/fuse/src/main/java/alluxio/fuse/options/FuseOptions.java:71]: Should remove JNR-FUSE but it's not removed yet

File: core/client/fs/src/main/java/alluxio/client/file/FileSystem.java (hunk 1 above)
Comment (LuQQiu) [reply core/client/fs/src/main/java/alluxio/client/file/FileSystem.java:165]: Currently it's the FileSystem options and is only supported to be enabled via FUSE, but can be coded to enable in other paths.
Not able to used via configuration through other alluxio clients yet

**Comment by LuQQiu** [comment]
alluxio-bot, feature merge this please

**Comment by LuQQiu** [comment]
alluxio-bot, feature-merge this please

**Pull request merged**